### ISSUE TITLE
Inline object in `anyOf`, `allOf` or `oneOf` not allowed

### ISSUE THREAD

**Issue by mbarnach** [mattpolzin/OpenAPIKit#98]
The following schema, extracted from the [OpenAPI documentation](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) is considered as invalid due to the inline object in the `allOf`.

```
openapi: 3.0.0
info:
    title: Sample API
    description: Optional multiline or single-line description in [CommonMark](http://commonmark.org/help/) or HTML.
    version: 0.1.9
servers:
    - url: http://api.example.com/v1
        description: Optional server description, e.g. Main (production) server
    - url: http://staging-api.example.com
        description: Optional server description, e.g. Internal staging server for testing
paths:
    /pets:
        patch:
            requestBody:
                content:
                    application/json:
                        schema:
                            oneOf:
                                - $ref: '#/components/schemas/Cat'
                                - $ref: '#/components/schemas/Dog'
                            discriminator:
                                propertyName: pet_type
            responses:
                '200':
                    description: Updated
components:
    schemas:
        Pet:
            type: object
            required:
                - pet_type
            properties: #3
                pet_type:
                    type: string
            discriminator:
                propertyName: pet_type
        Dog:
            allOf:
                - $ref: '#/components/schemas/Pet'
                - type: object # <== Allowed by OpenAPI but not validated.
                    properties:
                        bark:
                            type: boolean
                        breed:
                            type: string
                            enum: [Dingo, Husky, Retriever, Shepherd]
        Cat:
            allOf:
                - $ref: '#/components/schemas/Pet'
                - type: object
                    properties:
                        hunts:
                            type: boolean
                        age:
                            type: integer
```

**Comment by mattpolzin** [comment]
Thanks for the report! If I've successfully reproduced your problem (using your helpful example) then I believe the underlying problem is actually that I did not make schema fragments (which are used by the `allOf` schema) able to represent references to other schemas. In other words, the problem is actually the reference, not the inline schema.

Regardless, I've got a fix in the works.

**Comment by mattpolzin** [comment]
@mbarnach This should be fixed in v1.4.3. Please let me know if I've missed anything.

**Comment by mbarnach** [comment]
It is working indeed! Thanks for the reactivity! 🎉
